art_node.go

```diff
@@ -45,19 +45,21 @@ type ArtNode struct {
 	size      uint8
 
 	// Leaf Node Attributes
-	key      []byte
-	keySize  uint64
-	value    interface{}
-	nodeType uint8
+	key         []byte
+	keySize     uint64
+	completeKey []byte
+	value       interface{}
+	nodeType    uint8
 }
 
-func NewLeafNode(key []byte, value interface{}) *ArtNode {
+func NewLeafNode(key, completeKey []byte, value interface{}) *ArtNode {
```

Comment:
It is used a lot internally, and not always with the complete key.
I think the right thing to do is to convert the existing function to a private one and have the public one call that one with a duplicate.
Will fix soon.